tests/api/batch/test_utils.py

```diff
@@ -0,0 +1,176 @@
+    batch_tiles_mock.side_effect = tiles_sequence
+
+    sleep_mock = mocker.patch("time.sleep")
+    logging_mock = mocker.patch("logging.Logger.info")
+
+    results = monitor_batch_job("mocked-request", config=config, sleep_time=sleep_time)
+
+    assert isinstance(results, defaultdict)
+    assert set(results) == {BatchTileStatus.PROCESSED, BatchTileStatus.FAILED}
+    assert sum(len(tiles) for tiles in results.values()) == tile_count
+    for tile_status in [BatchTileStatus.PROCESSED, BatchTileStatus.FAILED]:
+        assert len(results[tile_status]) == tile_status_sequence[-1].get(tile_status, 0)
+
+    assert monitor_analysis_mock.call_count == 1
+
+    progress_loop_counts = len(tile_status_sequence) - 1
+
+    assert batch_tiles_mock.call_count == progress_loop_counts + 1
+    assert all(call.args == (batch_request,) and call.kwargs == {} for call in batch_tiles_mock.mock_calls)
+
+    assert sleep_mock.call_count == progress_loop_counts
+    assert all(call.args == (sleep_time,) and call.kwargs == {} for call in sleep_mock.mock_calls)
+
+    is_processing_logged = batch_status is BatchRequestStatus.PROCESSING
+    is_failure_logged = BatchTileStatus.FAILED in tile_status_sequence[-1]
+    assert logging_mock.call_count == int(is_processing_logged) + int(is_failure_logged)
+
+
+def _tile_status_counts_to_tiles(tile_status_counts: Dict[BatchTileStatus, int]) -> List[Dict[str, str]]:
+    """From the info about how many tiles should have certain status it generates a list of tile payloads with these
+    statuses.
+
+    Each payload should be approximately what Sentinel Hub service returns but because we don't need all parameters we
+    just return status. At the end we randomly shuffle the list just to make it more general.
+    """
+    tiles: List[Dict[str, str]] = []
+    for tile_status, count in tile_status_counts.items():
+        for _ in range(count):
+            tiles.append({"status": tile_status.value})
+
+    random.shuffle(tiles)
+    return tiles
+
+
+def test_monitor_batch_job_sleep_time_error() -> None:
+    with pytest.raises(ValueError):
+        monitor_batch_job("x", sleep_time=59)
+
+    with pytest.raises(ValueError):
+        monitor_batch_job("x", analysis_sleep_time=4)
+
+
+@pytest.mark.skipif(sys.version < "3.8", reason="Mocking check for call.args doesn't work correctly for Python 3.7")
+@pytest.mark.parametrize(
+    "status_sequence",
+    [
+        (BatchRequestStatus.CREATED, BatchRequestStatus.ANALYSING, BatchRequestStatus.ANALYSIS_DONE),
+        (BatchRequestStatus.DONE,),
+        (BatchRequestStatus.ANALYSING, BatchRequestStatus.CANCELED),
+        (BatchRequestStatus.FAILED,),
+    ],
+)
+@pytest.mark.parametrize("config", [SHConfig(), None])
+@pytest.mark.parametrize("sleep_time", [5, 1000])
+def test_monitor_batch_analysis(
+    status_sequence: Tuple[BatchRequestStatus, ...], config: SHConfig, sleep_time: int, mocker: MockerFixture
```

Comment:
why use tuples here?

Reply:
I guess I could switch to `Sequence` if you would prefer that. I chose `Tuple` just because it is immutable structure.

Reply:
Nevermind, at first look i thought there were only singleton tuples in the parametrization. Leave it as tuples